addons/mrp/tests/test_warehouse_multistep_manufacturing.py

```diff
@@ -149,11 +149,13 @@ def test_manufacturing_flow(self):
             'location_id': self.warehouse.wh_output_stock_loc_id.id,
             'location_dest_id': self.customer_location,
             'procure_method': 'make_to_order',
+            'origin': 'SOURCEDOCUMENT',
             'state': 'draft',
         })
         picking_customer.action_confirm()
         production_order = self.env['mrp.production'].search([('product_id', '=', self.finished_product.id)])
         self.assertTrue(production_order)
+        self.assertEqual(production_order.origin, 'SOURCEDOCUMENT', 'Should carry over origin')
 
         picking_stock_preprod = self.env['stock.move'].search([
```

Comment:
You also should probably check that the pre-prod picking `origin` is the same than the MO `name`:

```py
self.assertEqual(picking_stock_preprod.origin, production_order.name, 'The pre-prod origin should be the MO name')
```

And the post-prod has the source document as origin:
```py
self.assertEqual(picking_stock_postprod.origin, 'SOURCEDOCUMENT', 'The pre-prod origin should be the MO name')
```

This is not directly related to this fix, but that would ensure consistency in the future.

Reply:
And we could also check that:
```py
self.assertNotEqual(production_order.name, 'SOURCEDOCUMENT', 'The MO name should not be the origin of the move')
```

Reply:
Thanks!  Updated the tests.  For the 3-product case, I tested manually and got the expected result:
- MO1, for Product A, created with the SO name as source (pre-prod picking with MO1 name as source, post-prod picking with SO name as source)
- MO2, for Product B, created with the MO1 name as source (pre-prod picking with MO2 name as source, post-prod picking with MO1 name as source)